src/mono/wasm/Wasm.Build.Tests/TestAppScenarios/MemoryTests.cs

```diff
@@ -29,26 +29,49 @@ public async Task AllocateLargeHeapThenRepeatedlyInterop_NoWorkload() =>
     public async Task AllocateLargeHeapThenRepeatedlyInterop()
     {
         string config = "Release";
-        CopyTestAsset("WasmBasicTestApp", "MemoryTests", "App");
+        ProjectInfo info = CopyTestAsset(config, false, "WasmBasicTestApp", "MemoryTests", "App");
+        bool isPublish = false;
         string extraArgs = "-p:EmccMaximumHeapSize=4294901760";
-        BuildProject(config, assertAppBundle: false, extraArgs: extraArgs, expectSuccess: BuildTestBase.IsUsingWorkloads);
+        BuildTemplateProject(info,
+            new BuildProjectOptions(
+                info.Configuration,
+                info.ProjectName,
+                BinFrameworkDir: GetBinFrameworkDir(info.Configuration, isPublish),
+                // using EmccMaximumHeapSize forces native rebuild
+                ExpectedFileType: GetExpectedFileType(info, isPublish: isPublish, isNativeBuild: true),
+                IsPublish: isPublish,
+                ExpectSuccess: BuildTestBase.IsUsingWorkloads
+            ),
+            extraArgs: extraArgs
+        );
 
         if (BuildTestBase.IsUsingWorkloads)
         {
-            await RunSdkStyleAppForBuild(new (Configuration: config, TestScenario: "AllocateLargeHeapThenInterop"));
+            await RunForBuildWithDotnetRun(new (Configuration: config, TestScenario: "AllocateLargeHeapThenInterop"));
         }
     }
 
     [Fact]
     public async Task RunSimpleAppWithProfiler()
     {
         string config = "Release";
-        CopyTestAsset("WasmBasicTestApp", "ProfilerTest", "App");
+        ProjectInfo info = CopyTestAsset(config, false, "WasmBasicTestApp", "ProfilerTest", "App");
+        bool isPublish = false;
         // are are linking all 3 profilers, but below we only initialize log profiler and test it
         string extraArgs = $"-p:WasmProfilers=\"aot+browser+log\" -p:WasmBuildNative=true";
-        BuildProject(config, assertAppBundle: false, extraArgs: extraArgs);
+        BuildTemplateProject(info,
+            new BuildProjectOptions(
+                info.Configuration,
+                info.ProjectName,
+                BinFrameworkDir: GetBinFrameworkDir(info.Configuration, isPublish),
+                ExpectedFileType: GetExpectedFileType(info, isPublish: isPublish),
+                IsPublish: isPublish,
```

Comment:
I think publish should have separate options (maybe mostly inherited, but still)